src/GameLogic/PlugIns/GoldenInvasionPlugIn.cs

```diff
@@ -0,0 +1,390 @@
+﻿// <copyright file="GoldenInvasionPlugIn.cs" company="MUnique">
+// Licensed under the MIT License. See LICENSE file in the project root for full license information.
+// </copyright>
+
+namespace MUnique.OpenMU.GameLogic.PlugIns;
+
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Microsoft.Extensions.Logging;
+using MUnique.OpenMU.GameLogic.Attributes;
+using MUnique.OpenMU.GameLogic.MiniGames;
+using MUnique.OpenMU.GameLogic.NPC;
+using MUnique.OpenMU.GameLogic.Views;
+using MUnique.OpenMU.GameLogic.Views.World;
+using MUnique.OpenMU.PlugIns;
+
+/// <summary>
+/// Describe the state of invasion.
+/// </summary>
+public enum MobsInvasionState : byte
+{
+    NotStarted,
+    Initialized,
+    Started,
+}
+
+public class PeriodicInvasionConfiguration
+{
+    public bool IsActive { get; set; }
+    public List<TimeOnly> Timetable { get; set; } = new();
+    public int EventDuration { get; set; }
+    public int PreStartMessageDelay { get; set; }
+    public string? Message { get; set; }
+
+    /// <summary>
+    /// Check if current time is OK for starting an invasion.
+    /// </summary>
+    /// <returns>Returns true if the invasion can be started.</returns>
+    public bool IsRightTime()
+    {
+        if (this.Timetable.Count == 0)
+        {
+            return true;
+        }
+
+        var nowTime = TimeOnly.FromDateTime(DateTime.UtcNow);
+        var erlier = nowTime.Add(TimeSpan.FromSeconds(-5));
```

Comment:
Any idea for the simpler way?

Reply:
I think such a logic belongs into the GoldenInvasionPlugin. For example, you could remember the time of the last run, and if it didn't run for the configured time yet, it should start

Reply:
WhiteWizard and others events should works exactly so I want to extract it, maybe move it to abstract class.